lucene/facet/src/java/org/apache/lucene/facet/taxonomy/directory/DirectoryTaxonomyReader.java

```diff
@@ -322,9 +324,14 @@ public FacetLabel getPath(int ordinal) throws IOException {
         return res;
       }
     }
-    
-    Document doc = indexReader.document(ordinal);
-    FacetLabel ret = new FacetLabel(FacetsConfig.stringToPath(doc.get(Consts.FULL)));
+
+    int readerIndex = ReaderUtil.subIndex(ordinal, indexReader.leaves());
+    LeafReader leafReader = indexReader.leaves().get(readerIndex).reader();
+    BinaryDocValues values = leafReader.getBinaryDocValues(Consts.FULL);
```

Comment:
Could you open a new issue to optimize this better in the future, to do bulk lookup of `ordinal -> FacetLabel`?  And then add a `// TODO ` pointing to that issue and describing the optimization?

Reply:
Could you give an example of what that bulk lookup function params would look like / some reference in the code where something of this sort is implemented?

Reply:
I created a new issue [LUCENE 9476](https://issues.apache.org/jira/browse/LUCENE-9476#)